src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java

```diff
@@ -148,20 +148,29 @@ public boolean isPinned()
     @Override
     public RestAction<Void> pin()
     {
+        if (isEphemeral())
+            throw new IllegalStateException("Cannot pin this message as it is ephemeral.");
```

Comment:
I would suggest a better wording:

```suggestion
            throw new IllegalStateException("Cannot pin ephemeral messages");
```